src/examples/train_with_mixture.py

```diff
@@ -0,0 +1,242 @@
+"""
```

Comment:
As far as I can tell the only difference between this example and the other one is the dataset? If so I don't think it's worth having this full example. Instead I would recommend just adding usage examples to the docstrings of these new classes.

Reply:
Mainly worried that adding more example scripts will make them harder to maintain in the long run.